x-pack/plugins/infra/server/lib/log_analysis/log_entry_rate_analysis.ts

```diff
@@ -127,3 +143,130 @@ export async function getLogEntryRateBuckets(
     }
   }, []);
 }
+
+export async function getLogEntryRateExamples(
+  context: RequestHandlerContext & { infra: Required<InfraRequestHandlerContext> },
+  sourceId: string,
+  startTime: number,
+  endTime: number,
+  dataset: string,
+  exampleCount: number,
+  sourceConfiguration: InfraSource,
+  callWithRequest: KibanaFramework['callWithRequest']
```

Comment:
Would there be a downside to using `context.core.elasticsearch.legacy.client.callAsCurrentUser` instead of this?

Reply:
If I'm honest I'm not sure - I noticed that with the Kibana framework `callWithRequest` we do a little bit of preprocessing to the ES params, before calling `callAsCurrentUser`, things like:

```
 const includeFrozen = await uiSettings.client.get(UI_SETTINGS.SEARCH_INCLUDE_FROZEN);
    if (endpoint === 'msearch') {
      const maxConcurrentShardRequests = await uiSettings.client.get(
        UI_SETTINGS.COURIER_MAX_CONCURRENT_SHARD_REQUESTS
      );
      if (maxConcurrentShardRequests > 0) {
        params = { ...params, max_concurrent_shard_requests: maxConcurrentShardRequests };
      }
    }
```

this seemed important, so I opted to use that version. Is it safe to bypass the shared settings and so on?

Reply:
That's true... so I'm not respecting that when fetching the category examples :grimacing: I guess we should make a `search` function available via the context at some point that does something similar to our framework function.

So this is technically more correct right now :+1:

Reply:
> I guess we should make a search function available via the context at some point that does something similar to our framework function.

Yep, sounds good.